### PR TITLE
add build docker image and flist workflow

### PR DESCRIPTION
**Pull request by hossnys**
https://github.com/threefoldtech/threefold_hub/issues/66

### CODE REVIEW

File: .github/workflows/build_image.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        - uses: actions/checkout@v2
+        - uses: docker/build-push-action@v1
+          with:
+            path: docker/validator
+            repository: ${{ secrets.DOCKERHUB_USERNAME }}/tf-hub
+            username: ${{ secrets.DOCKERHUB_USERNAME }}
+            password: ${{ secrets.DOCKERHUB_PASSWORD }}
+            dockerfile: docker/validator/Dockerfile
+            tags: latest
+            build_args: BRANCH=development
```
Comment (sameh-farouk) [inline .github/workflows/build_image.yml]: is that needed? I can't see any ARG used in the Dockerfile.

File: .github/workflows/build_image.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    types: [published]
+  workflow_dispatch:
+
+  
+jobs:
+
+    build-docker:
+        runs-on: ubuntu-latest
+        steps:
+        - uses: actions/checkout@v2
```
Comment (sameh-farouk) [inline .github/workflows/build_image.yml:14]: why use deprecated build-push-action@v1, you probably need to use v2
you will have to modify the action inputs a bit, `path` -> `context`, and also don't forget to add `push` input as it defaults to false in the newer versions.

File: .github/workflows/build_image.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        - uses: docker/build-push-action@v1
+          with:
+            path: docker/validator
+            repository: ${{ secrets.DOCKERHUB_USERNAME }}/tf-hub
+            username: ${{ secrets.DOCKERHUB_USERNAME }}
+            password: ${{ secrets.DOCKERHUB_PASSWORD }}
+            dockerfile: docker/validator/Dockerfile
```
Comment (sameh-farouk) [inline .github/workflows/build_image.yml]: do we need this? it defaults to `{path}/Dockerfile` or on newer version  {context}/Dockerfile` so we can drop this line

**Review by sameh-farouk — CHANGES_REQUESTED** [review]

File: .github/workflows/build_image.yml (hunk 1 above)
Comment (hossnys) [reply .github/workflows/build_image.yml]: removed this line https://github.com/threefoldtech/threefold_hub/pull/70/commits/9e3bac9ad19c8ddf8787e532b633a37255556b06

File: .github/workflows/build_image.yml (hunk 2 above)
Comment (hossnys) [reply .github/workflows/build_image.yml:14]: changed to v2 https://github.com/threefoldtech/threefold_hub/pull/70/commits/9e3bac9ad19c8ddf8787e532b633a37255556b06

File: .github/workflows/build_image.yml (hunk 3 above)
Comment (hossnys) [reply .github/workflows/build_image.yml]: dropped this line https://github.com/threefoldtech/threefold_hub/pull/70/commits/9e3bac9ad19c8ddf8787e532b633a37255556b06

**Review by sameh-farouk — APPROVED** [review]

**Review by OmarElawady — APPROVED** [review]

**Pull request merged**